skimage/color/colorconv.py

```diff
@@ -804,6 +804,65 @@ def rgb2grey(rgb):
     return rgb2gray(rgb)
 
 
+def gray2rgba(image, alpha=None):
+    """Create a RGBA representation of a gray-level image.
+
+    Parameters
+    ----------
+    image : array_like
+        Input image.
+    alpha : array_like, optional
+        Alpha channel of the output image. It may be a scalar or an
+        array with the same shape as ``image``. If not specified it is
+        set to the maximum limit correspondin to ``image`` data type.
+
+    Returns
+    -------
+    rgba : ndarray
+        RGBA image. A new dimension of length 4 is added to input
+        image shape.
+
+    Raises
+    ------
+    ValueError
+        If ``alpha`` is a scala and is not in ``image`` data type
+        limits.
+    ValueError
+        If ``alpha`` is an array and have a different shape then
+        ``image``.
+    ValueError
+        If ``alpha`` is an array and have a different data type then
+        ``image``.
+
+    """
+
+    arr = np.asarray(image)
+
+    alpha_min, alpha_max = dtype_limits(arr, clip_negative=False)
+
+    if alpha is None:
+        alpha = alpha_max
+    elif np.isscalar(alpha):
```

Comment:
Again, I would simply remove all these checks.

Checking so often will cause headaches. The core functionality, broadcasting and adding a channel at the end remains the same, even if the user provides `alpha=2.0` for floats.